File: user/src/main/scala/ru/tdd/app/models/dto/UpdateUserDto.scala
package ru.tdd.app.models.dto

import ru.tdd.database.entities.users.AppUser
import zio.json.{DeriveJsonDecoder, DeriveJsonEncoder}

import java.time.{LocalDate, LocalDateTime}
import java.util.UUID

/**
 * @author Tribushko Danil
 * @since 20.11.2025
 * Dto запроса для обновления пользователя
 */
case class UpdateUserDto(
                          chatId: Long,
                          username: String,
                          birthday: LocalDate
                        )

object UpdateUserDto {

  implicit val encoder = DeriveJsonEncoder.gen[UpdateUserDto]
  implicit val decoder = DeriveJsonDecoder.gen[UpdateUserDto]
}

/**
 * @author Tribushko Danil
 * @since 20.11.2025
 * Dto пользователя
 */
case class UserDto(
                  id: UUID,
                  chatId: Long,
                  username: String,
                  birthday: LocalDate,
                  creationTime: LocalDateTime,
                  updateTime: LocalDateTime,
                  roles: Seq[BaseDTO]
                  )

object UserDto {

  implicit val encoder = DeriveJsonEncoder.gen[UserDto]
  implicit val decoder = DeriveJsonDecoder.gen[UserDto]

  def fromEntity(user: AppUser): UserDto =
    UserDto(
      user.id,
      user.chatId,
      user.username,
      user.birthday,
      user.creationTime,
      user.updateTime,
      user.roles.map(r => BaseDTO(r.name))
    )
}

File: telegram-bot/src/main/java/ru/tdd/telegram_bot/model/dto/users/UpdateUserDTO.java
package ru.tdd.telegram_bot.model.dto.users;

import ru.tdd.telegram_bot.controller.annotations.LocalDateJsonFormat;

import java.time.LocalDate;

/**
 * @author Tribushko Danil
 * @since 29.12.2025
 * DTO запроса на обновление пользователя
 */
public class UpdateUserDTO {

    private String username;

    @LocalDateJsonFormat
    private LocalDate birthday;

    public UpdateUserDTO(){}

    public UpdateUserDTO(String username, LocalDate birthday) {
        this.username = username;
        this.birthday = birthday;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public LocalDate getBirthday() {
        return birthday;
    }

    public void setBirthday(LocalDate birthday) {
        this.birthday = birthday;
    }
}
